examples/moonolith/ex1p.cpp

```diff
@@ -5,7 +5,8 @@
 // Moonolith sample runs:
```

Comment:
I noticed there is no `#ifdef MFEM_USE_MOONOLITH ...`, which we usually do for examples depending on external libraries.

Reply:
Added the check to every example